src/index.js

```diff
@@ -143,8 +158,13 @@ class CookieConsent extends Component {
     // remove listener if set
     window.removeEventListener("scroll", this.handleScroll);
 
-    if (setDeclineCookie) {
-      Cookies.set(cookieName, declineCookieValue, { expires: expires, ...extraCookieOptions });
+    if (setDeclineCookie && sameSite === SAMESITE_OPTIONS.NONE) {
```

Comment:
Please extract this logic to a private function, it's the same logic (with different params) from the accept version